Night mode

Fodilahy-mena opened the pull request:
Hello team!
I have added some functionalities to the night mode tool button for switching between dark mode and default mode with entire app theme colors and background. Please, have a look at it when you are available and leave comments if necessary.

- Branch link: https://github.com/onja-org/malagasy-rn-Polonium/tree/night-mode
- Click up link: https://app.clickup.com/t/35bjc2
- Refs: U11 #35bjc2

Thanks in advance

rinonTen commented:
You'll not be able to merge this branch after our approvals because you still have conflicts to resolve. 
Good work though!

rinonTen commented on src/components/Textarea/Textarea.js:
```diff
@@ -1,14 +1,40 @@
       <TextInput
-        style={editable ? styles.input : styles.textarea}
+        placeholderTextColor={
+          textColor === '#111827' ? 'rgba(17, 24, 39, 0.5)' : textColor
```
It would be more readable and efficient if you have variables for the colors because you're repeating the same color a few times.

Fodilahy-mena replied:
True!

rinonTen commented on src/components/Textarea/Textarea.js:
```diff
@@ -1,14 +1,40 @@
+            ? [
+                styles.input,
+                {
+                  color: textColor ? textColor : '#111827',
```
This one

andrzej-onja replied:
color strings should be stored as consts as they are repeating so many times '#111827'

rinonTen commented on src/components/Textarea/Textarea.js:
```diff
@@ -1,14 +1,40 @@
+                  color: textColor ? textColor : '#111827',
+                },
+              ]
+            : [styles.textarea, {color: textColor ? textColor : '#111827'}]
```
The same here

Fodilahy-mena replied:
This is the only way of approaching it. Maybe you do not understand the logic here. I am doing this because, for the components that do not have ` textColor` that passed to it, they use `"#111827"` color as a default. I what you mean about creating a variable but I don't think it's necessary.

rinonTen commented on src/redux/theme/index.js:
```diff
@@ -0,0 +1,24 @@
+      iconFill: '#111827',
+    },
+  },
+};
```
This is really good but I would use variables for those colors. It's not necessary though

Fodilahy-mena replied:
I would not use variables to assign to an object prop value, especially in this case.

andrzej-onja commented on src/containers/HomeScreenContainer.js:
```diff
@@ -1,28 +1,27 @@
-    currentCategoryId: currentCategoryIdRoot(state),
-    learntPhrases: learntPhrases(state),
-    isLearntPhrases: isLearntPhrases(state),
+    newTerms: newTermsRoot(state),
```
newTerms ?

andrzej-onja commented on src/redux/actions/index.js:
```diff
@@ -41,53 +40,51 @@ export function setLanguageName(language) {
+    type: GET_THEME_MODE,
+    payload: CUSTOM_THEMES.nightMode,
   };
 }
```
remove and just use set

andrzej-onja commented on src/redux/reducers/index.js:
```diff
@@ -47,22 +48,33 @@ function nativeLanguage(state = '', action) {
+function themeMode(state = {}, action) {
+  switch (action.type) {
+    case GET_THEME_MODE:
+      return action.payload;
```
remove get

andrzej-onja commented on src/redux/actions/index.js:
```diff
@@ -41,53 +40,51 @@ export function setLanguageName(language) {
+    payload: newTerms,
   };
 }
 
```
export function switchTheme() {
  return {
    type: SWITCH_THEME,
  };
}

andrzej-onja commented on src/redux/reducers/index.js:
```diff
@@ -47,22 +48,33 @@ function nativeLanguage(state = '', action) {
+    case GET_THEME_MODE:
+      return action.payload;
+    case SET_THEME_MODE:
+      return action.payload;
```
case  SWITCH_THEME:
return state ===CUSTOM_THEMES.nightMode
                        ? CUSTOM_THEMES.defaultMode
                        : CUSTOM_THEMES.nightMode

andrzej-onja commented on src/redux/theme/index.js:
```diff
@@ -0,0 +1,24 @@
+import {DefaultTheme, DarkTheme} from '@react-navigation/native';
```
try if it can be removed

andrzej-onja commented on src/redux/theme/index.js:
```diff
@@ -0,0 +1,24 @@
+import {DefaultTheme, DarkTheme} from '@react-navigation/native';
+
+export const CUSTOM_THEMES = {
+  defaultMode: {
```
add name or id prop : 'default' or 'dark'

andrzej-onja commented on src/screens/Home.js:
```diff
@@ -36,26 +38,42 @@ export default ({
 }) => {
+  useEffect(() => {
+    getThemeMode();
+  }, []);
```
set theme in initial state and remove this

Fodilahy-mena committed:
feat(Home, Learning, AddNewTerm screens): toggle between dark mode and default mode

_ Adds all of the functionalities for toggling between the two modes

_ Updates all of the components that need the properties from themeMode redux state

_ Adds theme folder to storing the CUSTOME_THEMES object

_ Adds action and reducer for themeMode

_ Updates tyles with some components that need themeMode's properties

_ Rebases night-mode branch to origin main

_ Fixes conflicts

_ Updates translation for learnt and seen phrases text

_ Gets rid of repeated string by using a variable

Refs: U11 #35bjc2

andrzej-onja commented on src/components/Textarea/Textarea.js:
```diff
@@ -1,14 +1,40 @@
       <TextInput
-        style={editable ? styles.input : styles.textarea}
+        placeholderTextColor={
+          textColor === '#111827' ? 'rgba(17, 24, 39, 0.5)' : textColor
```
this is weird `textColor === '#111827' ? 'rgba(17, 24, 39, 0.5)'`

Fodilahy-mena replied:
This is because we don't really have a specific color for placeholder in night mode.

andrzej-onja commented on src/components/Textarea/Textarea.js:
```diff
@@ -1,14 +1,40 @@
+  onChange = () => null,
+  placeholder,
+  backgroundColor,
+  textColor,
```
maybe just use default value `'#111827'`
`textColor='textColor,'`
and you can get rid of 
`color: textColor ? textColor : '#111827'`
and just use
`color: textColor`

Fodilahy-mena committed:
Merge branch 'main' into night-mode
